docs/src/userguide/language_basics.rst

```diff
@@ -169,7 +169,7 @@ C array can be declared by adding ``[ARRAY_SIZE]`` to the type of variable:
             def func():
                 g: cython.float[42]
                 f: cython.int[5][5][5]
-                ptr_char_array: cython.pointer(cython.char[4])  # pointer to the array of 4 chars
+                ptr_char_array: cython.pointer[cython.char[4]]  # pointer to the array of 4 chars
```

Comment:
mypy is actually unhappy with the `char[4]` notation, which is understandable given that `4` is not a type and the `char` (read: `py_int`) type doesn't allow indexing. Also not something to resolve in this PR.